okhttp-logging-interceptor/src/main/kotlin/okhttp3/logging/HttpLoggingInterceptor.kt

```diff
@@ -163,29 +200,90 @@ class HttpLoggingInterceptor
 
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
-      val level = this.level
+      var request = chain.request()
+
+      val level = this.requestLevel(request)
 
-      val request = chain.request()
       if (level == Level.NONE) {
         return chain.proceed(request)
       }
 
-      val logBody = level == Level.BODY
-      val logHeaders = logBody || level == Level.HEADERS
+      val logHeaders = level == Level.BODY || level == Level.HEADERS
 
-      val requestBody = request.body
+      val decompressionAlgorithms = findCompressionAlgorithms(chain)
 
       val connection = chain.connection()
+
+      val requestBody = request.body
       var requestStartMessage =
         ("--> ${request.method} ${redactUrl(request.url)}${if (connection != null) " " + connection.protocol() else ""}")
-      if (!logHeaders && requestBody != null) {
+      if (!logHeaders && level != Level.STREAMING && requestBody != null) {
         requestStartMessage += " (${requestBody.contentLength()}-byte body)"
       }
       logger.log(requestStartMessage)
 
-      if (logHeaders) {
-        val headers = request.headers
+      if (level != Level.BASIC) {
+        request = logRequest(request, level, decompressionAlgorithms)
+      }
+
+      val startNs = System.nanoTime()
+      val response: Response
+      try {
+        response = chain.proceed(request)
+      } catch (e: Exception) {
```

Comment:
I think the better control flow for this is different:

```
var success = false
try {
  ...
  success = true
} finally {
  if (!success) { ... }
}
```

As-is you’re not catching stuff like `AssertionError` which I think is unintentional?

Reply:
(hang on, this is just moved code. Sorry for me reviewing old code!)